### PR TITLE
add correct value when cloning http headers

### LINKED ISSUE #35497: DefaultRequestHeaders are missing in request when added with TryAddWithoutValidation
This is regression caused by #34724 and discussed in #35480.
This seems to be used by Azure storage code and the request are failing. 
When used with debug version, it hits assert in WriteHeadersAsync()

### PR DESCRIPTION
**Pull request by wfurt**
fixes #35497

### CODE REVIEW

**Comment by ViktorHofer** [comment]
Thanks for following up on this so fast 👍

File: src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Headers.cs [hunk 1]
```diff
@@ -45,6 +45,33 @@ public async Task SendAsync_UserAgent_CorrectlyWritten()
             });
         }
 
+        [Fact]
+        public async Task SendAsync_DefaultHeaders_CorrectlyWritten()
+        {
+            string version = "2017-04-17";
+            string blob = "BlockBlob";
```
Comment (stephentoub) [inline src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Headers.cs:52]: Nit: const and PascalCasing

**Review by stephentoub — APPROVED** [review]
Nice job tracking this down. Did my typo make it into a preview branch? If yes we should port the fix.

**Comment by stephentoub** [comment]
Sorry for my typo.  Nice job tracking this down.

**Comment by stephentoub** [comment]
Unfortunately it looks like my mistake made it into Preview 4.

**Review by ViktorHofer — APPROVED** [review]

**Comment by ViktorHofer** [comment]
> Unfortunately it looks like my mistake made it into Preview 4.

@stephentoub I guess we can still bring this to shiproom for P4?

**Comment by stephentoub** [comment]
Preview 4 shipped two weeks ago

**Comment by stephentoub** [comment]
Oh, wait, that way Preview 3?

**Comment by ViktorHofer** [comment]
following up on that with Stephen offline. We haven't shipped P4 yet.

**Comment by ViktorHofer** [comment]
Failures are https://github.com/dotnet/runtime/issues/35507 and remote executor, probably because of the machine load.

**Pull request merged**